pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -109,13 +110,25 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
         if instance.Spec.KubeVirtImagePullPolicy != "" {
            reqLogger := log.WithValues("imagePullPolicy", instance.Spec.KubeVirtImagePullPolicy)
            reqLogger.Info("HCO CR contains KubeVirt Image Pull Policy")
-           KubeVirtImagePullPolicy = instance.Spec.KubeVirtImagePullPolicy
+           _, err = verifyPullPolicyType(instance.Spec.KubeVirtImagePullPolicy)
+           if err != nil {
+               reqLogger := log.WithValues("HCO CR imagePullPolicy", instance.Spec.KubeVirtImagePullPolicy, "default imagePullPolicy", KubeVirtImagePullPolicy)
+               reqLogger.Info("HCO CR KubeVirt Pull Policy is invalid. Reverting to default policy")
```

Comment:
i don't see the code "reverting" the policy. also, shouldn't we fail the operation rather then try to "fix" it? what if i pass "Alwayss" and this defaults me to "IfNotPresent".

Reply:
How exactly you think we can propagate this failure back to the user?

Reply:
Status on the CR

Reply:
We basically have 2 options here: we either stop processing - which is actually the same as if no verification code would have been added, still leaving the user clueless as to what have happened, and why the Operators haven't been started - even though the HCO CR manifest was imported.

OR, we can verify the values and disregard invalid ones, backing off to the default values.
While still properly mentioning this in the log.

IMHO the second option is much better - the system is still functioning, and if the use of default policy still creates some issue then the user will look up the logs and find the message there.

Reply:
I don't understand. Putting failure status messages on the CR is a complete and totally valid response to an invalid spec.

Reply:
> 
> 
> I don't understand. Putting failure status messages on the CR is a complete and totally valid response to an invalid spec.

So you think that we'll actually be better off if we halt/stop the processing at this point?

From my point of view it's better to try and continue running if we can fix the situation, especially as we don't have an option to note the user directly during or just after the CR manifest was created - and our only options are posting something to log and/or CR status.

Also right now the CR is not re-processed on the change (edit), thus in case of i.e. typo/invalid policy one will need to actually delete the CR, fix the issue in the file and re-import again.

Reply:
Trying to be smart, in case of a failure, can lead to unexpected behaviour. If the configuration is wrong. Also, I can imagine it could be problematic once we need to upgrade to newer version of the spec. Something invalid used to work, but it does not anymore. I think we should be strict with this, but I won't block this PR.

Reply:
I think there are some additional risks we take by reverting to defaults.  For example, if one parameter fails, do we revert all parameters back to their defaults? Or only the one failed param?  If we only do the failed param, we run the risk if using a default that's not compatible with the user's imputed config.  We also aren't sure if our defaults will work in all environments.

IMO, we should report status on the CR like `invalid input for parameter ImagePullPolicy` and let the user take action.